From 505cbc78dd98f04915220e3d2796bc026f79c066 Mon Sep 17 00:00:00 2001
From: Michel Salim <michel@fb.com>
Date: Mon, 26 Oct 2020 16:30:30 -0700
Subject: [PATCH] allow overriding version

Summary:
Upstream doesn't currently guarantee ABI stability, we want to let
downstream distributions override the version themselves when they
update Fizz.

Reviewed By: danobi

Differential Revision: D24498696

fbshipit-source-id: 69d5b7d16ef890af8cdfa7950b07002be11ab7fb
---
 fizz/CMakeLists.txt | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/fizz/CMakeLists.txt b/fizz/CMakeLists.txt
index 5a67778a..a1ad30c6 100644
--- a/fizz/CMakeLists.txt
+++ b/fizz/CMakeLists.txt
@@ -3,7 +3,15 @@
 
 cmake_minimum_required(VERSION 3.1)
 
-project("fizz" VERSION 1.0.0 LANGUAGES CXX C)
+if (NOT DEFINED PACKAGE_VERSION)
+  set(PACKAGE_VERSION "1.0.0")
+endif()
+
+if (NOT DEFINED SO_VERSION)
+  set(SO_VERSION 1)
+endif()
+
+project("fizz" VERSION ${PACKAGE_VERSION} LANGUAGES CXX C)
 
 if (NOT DEFINED CPACK_GENERATOR)
   set(CPACK_GENERATOR "RPM")
@@ -208,7 +216,7 @@ add_library(fizz
 
 if (BUILD_SHARED_LIBS)
   set_target_properties(fizz
-    PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION 1)
+    PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${SO_VERSION})
 endif()
 
 get_filename_component(FIZZ_BASE_DIR ${CMAKE_SOURCE_DIR}/.. ABSOLUTE)
